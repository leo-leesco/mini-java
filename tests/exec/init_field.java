class A {
    String x;
    int y;
    boolean z;
    A next;
    A() {}
}
class Main {
    public static void main(String args[]) {
        A a = new A();
        if (a.x == null)
            System.out.print("x = null\n");
        System.out.print("y = " + a.y + "\n");
        if (!a.z)
            System.out.print("z = false\n");
        if (a.next == null)
            System.out.print("next = null\n");
    }
}

