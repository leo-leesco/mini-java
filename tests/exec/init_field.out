x = null
y = 0
z = false
next = null
